pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -80,16 +80,23 @@ func (k *KubeASCheck) Run() error {
 		k.Warn("Leader Election not enabled. Not running Kubernetes API Server check or collecting Kubernetes Events.")
 		return nil
 	}
-	err = k.runLeaderElection()
-	if err != nil {
+
+	errLeader := k.runLeaderElection()
+	if errLeader != nil && errLeader != apiserver.ErrNotLeader {
 		return err
 	}
 
-	// We start the API Server Client.
-	k.ac, err = apiserver.GetAPIClient()
-	if err != nil {
-		k.Warn("Could not connect to apiserver: %s", err)
-		return err
+	if k.ac == nil {
+		// We start the API Server Client.
+		k.ac, err = apiserver.GetAPIClient()
+		if err != nil {
+			k.Warn("Could not connect to apiserver: %s", err)
+			return err
+		}
+	}
+
+	if errLeader == apiserver.ErrNotLeader {
```

Comment:
Is `errLeader` changed between line 85 and there? 🤔

Reply:
If errLeader is not leader we want to run the service mapper, that is started when using GetAPIClient.
If the error is something else we want to throw the error even before